doc/schema.md

```diff
@@ -0,0 +1,75 @@
+application-compatible subscriptions and user-controllable subscriptions, this
+document describes necessary changes to ZeroEQ.
+
+Use-cases are coupling of applications that belong to a certain user/session,
+not coupling all applications on the same network. User-interfaces for selection
+of applications that should connect to each other need more information for the
+user and for the system to perform filtering.
+
+# Overview of changes
+
+* Fixed schema: user-specified schemas are ignored. Filtering is now done via
+  sessions and compatible events.
+* Introduction of Session: only publishers of the same session and compatible
+  events are automatically subscribed in the subscriber. The default session
+  name is the current username. It can be set to a different value at runtime
+  with the ZEROEQ_SESSION environment variable. If the session name is
+  empty/non-existant, no automatic subscription is performed.
+* The ZeroConf record contains information about published/subscribed events,
+  application name (derived automatically), username and session.
+* Vocabulary stores events from an associated publisher or subscriber. The
+  vocabulary information can be requested (this class implements request-reply
+  using the REQ-REP socket from zmq). One vocabulary is associated with exactly
+  one publisher or with one subscriber.
+* Broker can handle explicit (un)subscriptions instructed by a remote
+  application.
+
+# Changes for Publisher
+
+* ZeroConf/Servus service name is _zeroeq_pub._tcp
+* Schema is fixed, internal and ignored if specified
+* URI can contain hostname and port
+* ZeroConf announcement:
+** Instance name is hostname:port
+** txt record:
+*** vocabulary_request = zmq connection string
+*** vocabulary = semi-colon separated list of published event UUIDs
+*** session = <user-name> by default, env ZEROEQ_SESSION if specified
+*** user = <user-name>
+*** application = <application-name>
+* Published events must be registered in the publisher. Non-registered events
+  cannot be published.
+
+# Changes for Subscriber
+
+* ZeroConf/Servus service name is _zeroeq_sub._tcp
+* Schema is fixed, internal and ignored if specified
+* URI can contain hostname and port
+* Automatic subscription only via session (default session is username),
+  otherwise explicit subscription via Broker.
+* ZeroConf announcement:
+** Instance name is a UUID to uniquely identify this subscriber
+** txt record:
+*** vocabulary_request = zmq connection string
+*** vocabulary = semi-colon separated list of published event UUIDs
+*** session = <user-name> by default, env ZEROEQ_SESSION if specified
+*** user = <user-name>
+*** application = <application-name>
+
+# Changes for Broker
+
+* Supports subscribe/unsubscribe requests for any remote application
+
+# New class Vocabulary
+
+* Provides a request-reply for vocabulary information
+* Holds registered events from a publisher or subscriber
+* Accessible via getVocabulary() from publisher/subscriber
```

Comment:
Need way of having per-publisher event subscriptions, either using two subscribers or registerHandler( publisher, event, func );